src/builders/indexable-hierarchy-builder.php

```diff
@@ -132,7 +132,19 @@ public function build( Indexable $indexable ) {
 	 * @return bool True when indexable has a built hierarchy.
 	 */
 	protected function hierarchy_is_built( Indexable $indexable ) {
-		if ( \in_array( $indexable->id, $this->saved_ancestors, true ) ) {
+
+		/**
+		 * Filters the parameter to allow building hierarchy.
+		 *
+		 * Used when adding term with `wp_set_object_terms` together with `wp_insert_post`.
+		 *
+		 * @since 21.2
+		 *
+		 * @return bool The filtered value of the `in_saved_ancestors` parameter.
+		 */
+		$in_saved_ancestors = apply_filters( 'Yoast\WP\SEO\manual_set_object_terms', \in_array( $indexable->id, $this->saved_ancestors, true ) );
```

Comment:
I think it might be nice to just split this up in 2 parts. Keep the existing in_array about if it is already saved. And than above it do the filter. So something like:
```
/**
		 * Filters ignoring checking if the hierarchy is already built.
		 *
		 * Used when adding term with `wp_set_object_terms` together with `wp_insert_post`.
		 *
		 * @api bool If the hierarchy already saved check should be ignored.
		 * @since 21.2
		 *
		 * @return bool The filtered value of the `in_saved_ancestors` parameter.
		 */
		$ignore_already_saved = apply_filters( 'wpseo_hierarchy_ignore_already_saved', false );
		if ( $ignore_already_saved ) {
			return false;
		}
```